objective-c/src/Ice/SlicedDataI.h

```diff
@@ -4,13 +4,15 @@
 
 #import <objc/Ice/SlicedData.h>
 
+#include <Ice/InputStream.h>
+#include <Ice/OutputStream.h>
 #include <Ice/SlicedData.h>
 
 @interface ICESlicedData : NSObject<ICESlicedData>
 {
 @private
-    Ice::SlicedData* slicedData_;
+    Ice::SlicedDataPtr slicedData_;
```

Comment:
My understanding here:
we can use a C++ data member with a destructor because the type (shared_ptr) has no virtual function. And the cleanup of ICESlicedData will run the C++ destructors. Would be good to double-check with a small test.

In the previous version, we were using a pointer to the Ptr and made calls to __incRef and __decRef.